### PR TITLE
Generate theme page from YAML and theme page reorg

### LINKED ISSUE #764: Add new themed page for registration required
Add new pages here

Re: https://github.com/FusionAuth/fusionauth-issues/issues/439
Themed doc: https://site-local.fusionauth.io/docs/v1/tech/themes/

### LINKED ISSUE #765: Add new themed page for Start IdP Link
Add new pages here

New page for 1.28.0, `StartIdpLinkAction`
Themed doc: https://site-local.fusionauth.io/docs/v1/tech/themes/

### LINKED ISSUE #786: Audit theme template variables
https://fusionauth.io/docs/v1/tech/themes/

- check to ensure all template are listed and explained with the correct URL
- check to ensure all available variables are defined and documented. There may be variables listed that are incorrect, and there may be variables that are not listed. 

<hr></hr>

Check against the top of the FTL file.  For instance, below should have `application`, `user` and `tenant` in the doc.  Diff between doc and ftl templates
```
[#-- @ftlvariable name="application" type="io.fusionauth.domain.Application" --]
[#-- @ftlvariable name="client_id" type="java.lang.String" --]
[#-- @ftlvariable name="user" type="io.fusionauth.domain.User" --]
[#-- @ftlvariable name="tenant" type="io.fusionauth.domain.Tenant" --]
[#-- @ftlvariable name="tenantId" type="java.util.UUID" --]
```

### LINKED ISSUE #826: Verify themed doc pages includes IdPs for the register page in addition to the authorize page
- https://github.com/FusionAuth/fusionauth-issues/issues/554

### PR DESCRIPTION
**Pull request by jobannon**
# Summary

This PR generates the theme pages, including custom template variables, from a yaml file. It also reorgs the theme pages a bit to make them easier to navigate.

It also includes some updates that came from @jobannon 's theme audit, but that will be completed in a different PR.

# OLD Summary 
Audit theme pages and variables 

WIP, but the current state

       - [x] accountEdit
       - [❓] accountIndex
       - [❓] accountTwoFactorDisable
       - [x] accountTwoFactorEnable
       - [x] accountTwoFactorIndex
       - [x] emailComplete
       - [x] emailSent
       - [❓] emailVerificationRequired (todo)
       - [❓] emailVerify
       - [ ] helpers
       - [ ] index
       - [❓] oauth2Authorize
       - [ ] oauth2AuthorizedNotRegistered
       - [ ] oauth2ChildRegistrationNotAllowed
       - [ ] oauth2ChildRegistrationNotAllowedComplete
       - [ ] oauth2CompleteRegistration
       - [ ] oauth2Device
       - [ ] oauth2DeviceComplete
       - [ ] oauth2Error
       - [ ] oauth2Logout
       - [ ] oauth2Passwordless
       - [ ] oauth2Register
       - [ ] oauth2StartIdPLink
       - [ ] oauth2TwoFactor
       - [ ] oauth2TwoFactorMethods
       - [ ] oauth2Wait
       - [ ] passwordChange
       - [ ] passwordComplete
       - [ ] passwordForgot
       - [ ] passwordSent
       - [ ] registrationComplete
       - [ ] registrationSent
       - [ ] registrationVerificationRequired
       - [ ] registrationVerify
       - [ ] samlv2Logout
       - [ ] unauthorized

### Related PRs/Issues 
Closes #826 
Closes #786 
Closes #765 
Closes #764

### CODE REVIEW

**Comment by robotdan** [comment]
I think @mooreds may have just done this, but it probably won't hurt to do a double check to ensure everything is accounted for.

**Comment by mooreds** [comment]
So I did part of this. Here's the PR: https://github.com/FusionAuth/fusionauth-site/pull/886

That PR:
* updated the API doc (/doc/v1/tech/apis/themes)
* updated the template section under https://fusionauth.io/docs/v1/tech/themes/#create-a-theme 

What I didn't do was:

* audit variables available in themes
* add the page section under the https://fusionauth.io/docs/v1/tech/themes/#template-variables section

Hope this helps.

File: site/docs/v1/tech/themes/index.adoc [hunk 1]
```diff
@@ -9,27 +9,28 @@ description: Learn how to theme and the FusionAuth login pages (including forgot
 
 FusionAuth themes allow you to customize the OAuth2 / OpenID Connect login pages and other user workflows such as forgot password. In FusionAuth you may create one to many themes and assign a theme per tenant or application so that you can customize the user experience for different users.
 
-In this guide you will find a video demonstration, a list of each template available to you and some example code.
+To demonstrate what is possible with the FusionAuth feature we have created a video that illustrates how to make FusionAuth look like StackOverflow.
+
+video::_ro3jH5Xkgo[youtube,width=560,height=315]
+
+Here are the topics in this section:
```
Comment (jobannon) [inline site/docs/v1/tech/themes/index.adoc:16]: Maybe be we need a section here?  Something like 

== How to Navigate Our Themes Section

File: site/docs/v1/tech/themes/index.adoc [hunk 2]
```diff
@@ -9,27 +9,28 @@ description: Learn how to theme and the FusionAuth login pages (including forgot
...
+
+* link:examples[Examples]
+* link:helpers[Helpers]
+* link:localization[Localization]
+* link:template-variables[Template Variables]
+
```
Comment (jobannon) [inline site/docs/v1/tech/themes/index.adoc:22]: This could be a section maybe?

== How to Create, Update, and Troubleshoot Themes

**Review by jobannon — COMMENTED** [review]
@mooreds I added a few suggestions, but otherwise, this is a great change!

File: site/docs/v1/tech/themes/index.adoc (hunk 1 above)
Comment (mooreds) [reply site/docs/v1/tech/themes/index.adoc:16]: I pretty much copied what we have in Events and Webhooks: https://fusionauth.io/docs/v1/tech/events-webhooks/

Just added a link to the API as well.

File: site/docs/v1/tech/themes/index.adoc (hunk 2 above)
Comment (mooreds) [reply site/docs/v1/tech/themes/index.adoc:22]: hmmm. Not sure. I think it's cleaner without the additional section.

**Comment by mooreds** [comment]
Your variable changes look good to me, @jobannon !

**Comment by mooreds** [comment]
Merging. Next steps are for @jobannon to complete the theme audit with the new yaml datastructure.

**Pull request merged**